src/main/kotlin/com/amazon/elasticsearch/replication/task/shard/ShardReplicationTask.kt

```diff
@@ -226,12 +235,15 @@ class ShardReplicationTask(id: Long, type: String, action: String, description:
                         logDebug("pushed to sequencer $fromSeqNo-$toSeqNo")
                         changeTracker.updateBatchFetched(true, fromSeqNo, toSeqNo, changesResponse.changes.lastOrNull()?.seqNo() ?: fromSeqNo - 1,
                             changesResponse.lastSyncedGlobalCheckpoint)
+                        //reset backoff after every successful getChanges call
+                        backOffForRetry = initialBackoffMillis
                     } catch (e: ElasticsearchTimeoutException) {
                         logInfo("Timed out waiting for new changes. Current seqNo: $fromSeqNo. $e")
                         changeTracker.updateBatchFetched(false, fromSeqNo, toSeqNo, fromSeqNo - 1,-1)
                     } catch (e: NodeNotConnectedException) {
                         logInfo("Node not connected. Retrying request using a different node. ${e.stackTraceToString()}")
-                        delay(backOffForNodeDiscovery)
+                        delay(backOffForRetry)
+                        backOffForRetry = (backOffForRetry * factor).toLong().coerceAtMost(maxTimeOut)
```

Comment:
We should probably add a comment as to why we are not adding delay/backoff in case of ESTimeout

Reply:
We already wait 1 minute on the server side before TimeoutException is thrown, so we dont need to add delay again at follower to get changes